packages/harmony-core/src/harmonyExtension.ts

```diff
@@ -182,7 +182,7 @@ export class HarmonyExtension {
           transaction.setParams({
             ...transaction.txParams,
             from: crypto.getAddress(extensionAccount.address).bech32,
-            nonce: Number.parseInt(utils.hexToNumber(nonce.result), 10),
+            nonce: Number.parseInt(utils.isHex(nonce.result.toString()) ? utils.hexToNumber(nonce.result.toString()) : nonce.result.toString(), 10),
```

Comment:
I think it can be written as 
```
nonce: Number.parseInt(nonce.result.toString()),
```

Reply:
Yeah - but it's typically not recommended to not pass a `radix` param - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt:

> A value passed as the radix argument is coerced to a Number (if necessary), then if the value is 0, NaN or Infinity (undefined is coerced to NaN), JavaScript assumes the following:
> 
> - If the input string begins with "0x" or "0X" (a zero, followed by lowercase or uppercase X), radix is assumed to be 16 and the rest of the string is parsed as a hexadecimal number.
> - If the input string begins with "0" (a zero), radix is assumed to be 8 (octal) or 10 (decimal). Exactly which radix is chosen is implementation-dependent. ECMAScript 5 clarifies that 10 (decimal) should be used, but not all browsers support this yet. **For this reason, always specify a radix when using parseInt.**
> - If the input string begins with any other value, the radix is 10 (decimal).